build: adopt Magefile for running acceptance tests

mridang opened the pull request:
## Description

This PR swaps our bash-based test harness for a self-documenting Magefile. mage test:default now spins up the namespaced Docker-Compose stack, waits for health checks plus a 30-second buffer, runs the acceptance tests, streams compose logs automatically when debug flags are set, and then tears everything down. A test:cover variant produces profile.cov for Codecov. The change gives us a single, reproducible command that works identically on every developer machine and in CI while eliminating flaky start-up races and manual log collection.

## Motivation and Context

It simplifies running the tests locally.

## How Has This Been Tested?

By both running the tests locally and also by updating the workflows on GitHub.

## Documentation:

None.

## Checklist:

None.

### Definition of Ready

- [ ] Short description of the feature/issue is added in the pr description
- [ ] PR is linked to the corresponding user story
- [ ] Acceptance criteria are met
- [ ] All open todos and follow ups are defined in a new ticket and justified
- [ ] Deviations from the acceptance criteria and design are agreed with the PO and documented.
- [ ] No debug or dead code
- [ ] My code has no repetitions
- [ ] All non-functional requirements are met
- [ ] The generic lifecycle acceptance test passes for affected resources.
- [ ] Examples are up-to-date and meaningful. The provider version is incremented.
- [ ] Docs are generated.
- [ ] Code is generated where possible.

mridang committed:
build: adopt Magefile for running acceptance tests

eliobischof commented on Magefile.go:
```diff
@@ -0,0 +1,133 @@
+	if err := sh.Run("golangci-lint", "run", "./..."); err == nil {
+		return nil
+	}
+	fmt.Println("golangci-lint not found – falling back to `go vet`")
+	return sh.RunV("go", "vet", "./...")
```
Let's try to have reproducible results by requiring a specific version of golangci-lint pinned in go.mod?
https://go.dev/doc/go1.24#tools

eliobischof commented on Magefile.go:
```diff
@@ -0,0 +1,133 @@
+		"up", "--detach", "--wait"); err != nil {
+		return err
+	}
+	// Extra buffer so that flaky startup races are less likely.
+	if err := sh.RunV("sleep", "30"); err != nil {
+		fmt.Println("warning: sleep command failed:", err)
+	}
```
Is there a condition we could retry quering until it is satisfied?

eliobischof reviewed (changes requested):
In a go repo, Mage is definitely better than Make :-)

Let's update the CONTRIBUTING.md

mridang committed:
Merge branch 'main' into build/migrate-to-magefile

mridang committed:
chore: upgrade ZITADEL to v4.7.0 and fix SVG format

Updates SVG test fixtures to include XML declaration and namespace
attributes required by Zitadel v4.7.0's API validation.

mridang committed:
Merge branch 'chore/upgrade-zitadel-4.7.0' into build/migrate-to-magefile

mridang committed:
chore: updgrade to zitadel-go 3.19

mridang committed:
Merge branch 'chore/upgrade-zitadel-4.7.0' into build/migrate-to-magefile

mridang committed:
Merge branch 'main' into build/migrate-to-magefile

mridang committed:
Merge branch 'main' into build/migrate-to-magefile
